components/admin_console/custom_terms_of_service_settings/custom_terms_of_service_settings.tsx

```diff
@@ -15,27 +16,40 @@ import LoadingScreen from 'components/loading_screen';
 
 import {Constants} from 'utils/constants';
 
-export default class CustomTermsOfServiceSettings extends AdminSettings {
-    static propTypes = {
-        actions: PropTypes.shape({
-            getTermsOfService: PropTypes.func.isRequired,
-            createTermsOfService: PropTypes.func.isRequired,
-        }).isRequired,
-        config: PropTypes.object,
-        license: PropTypes.object,
-        setNavigationBlocked: PropTypes.func,
-
-        /*
-        * Action to save config file
-        */
-        updateConfig: PropTypes.func,
+type Props = BaseProps & {
+    actions: {
+        getTermsOfService: () => Promise<{data: TermsOfService}>;
+        createTermsOfService: (text: string) => Promise<{data: TermsOfService, error?: Error}>;
     };
+    config: DeepPartial<AdminConfig>;
+    license: ClientLicense,
+    setNavigationBlocked: () => void,
+
+    /*
+     * Action to save config file
+     */
+    updateConfig: () => void
+};
+
+type State = BaseState & {
+    termsEnabled: boolean;
+    reAcceptancePeriod: number;
+    loadingTermsText: boolean;
+    receivedTermsText: string;
+    termsText: string;
+    saveNeeded: boolean;
+    saving: boolean;
+    serverError: JSX.Element | string | null;
+    errorTooltip: boolean;
+}
 
-    constructor(props) {
+export default class CustomTermsOfServiceSettings extends AdminSettings<Props, State> {
+    constructor(props: Props) {
         super(props);
+
         this.state = {
-            termsEnabled: props.config.SupportSettings.CustomTermsOfServiceEnabled,
-            reAcceptancePeriod: props.config.SupportSettings.CustomTermsOfServiceReAcceptancePeriod,
+            termsEnabled: Boolean(props.config.SupportSettings?.CustomTermsOfServiceEnabled),
+            reAcceptancePeriod: this.parseIntNonZero(String(props.config.SupportSettings?.CustomTermsOfServiceReAcceptancePeriod), Constants.DEFAULT_TERMS_OF_SERVICE_RE_ACCEPTANCE_PERIOD),
```

Comment:
Does it make sense to keep termsEnabled and reAcceptancePeriod as possibly undefined (i.e. `termsEnabled?: boolean`)? I think it would simplify these two assignments and other future assignments.

Reply:
Can we look at how this assignment can be simplified? It seems to call for handling like Boolean(termsEnabled) and (reAcceptancePeriod || '') in more places.

Reply:
I wonder why just doing `termsEnabled: props.config.SupportSettings?.CustomTermsOfServiceEnabled` is not working. The same with the next line. Why is it necessary to call the `String(...)` and `Boolean(...)` functions here?

Reply:
I've pushed a commit marking these two props as optional which does clean up the constructor here. I think I was hesitant to go this route due to the other places where the casting is needed, after this change the overall does look cleaner.